src/aind_behavior_video_transformation/etl.py

```diff
@@ -64,6 +77,137 @@ class BehaviorVideoJob(GenericEtl[BehaviorVideoJobSettings]):
+                        override_path = input_dir / override_path
+                    for override_file in override_path.rglob("*"):
+                        if override_file.is_file() and (
+                            override_file.suffix.lower()
+                            in self.job_settings.video_extensions
+                        ):
+                            override_file = override_file.resolve()
+                            overrides[override_file] = override_req
+
+            # Filter path_req_pairs by overrides.
+            output_pairs: List[Tuple[Path, CompressionRequest]] = []
+            for vid_path, default_req in path_req_pairs:
+                for override_path, override_req in overrides.items():
+                    if vid_path.samefile(override_path):
+                        output_pairs.append((override_path, override_req))
+                    else:
+                        output_pairs.append((vid_path, default_req))
+            path_req_pairs = output_pairs
+
+        return path_req_pairs
+
+    def _run_compression(
+        self,
+        path_comp_req_pairs: List[Tuple[PathLike, CompressionRequest]],
+    ) -> None:
+        """
+        Runs CompressionRequests at the specified paths.
+        """
+        input_dir = Path(self.job_settings.input_source.resolve())
+        output_dir = Path(self.job_settings.output_directory.resolve())
+
+        convert_video_params = []
+        for vid_path, comp_req in path_comp_req_pairs:
+            # Resolve destination
+            relative_path = vid_path.relative_to(input_dir)
+            output_path = output_dir / relative_path
+            output_vid_dir = output_path.parent
+
+            # Resolve compression params
+            arg_set = comp_req.determine_ffmpeg_arg_set()
+
+            # Add to job buffer
+            convert_video_params.append((vid_path, output_vid_dir, arg_set))
+            logging.info(
+                f"Compressing {str(vid_path)} \
+                         w/ {comp_req.compression_enum}"
+            )
+
+        if self.job_settings.parallel_compression:
+            # Dask implementation
+            # import dask
+            # jobs = [dask.delayed(convert_video)(*params)
+            #         for params in convert_video_params]
+            # dask.compute(*jobs)  # This returns an error if any jobs fail
+
+            # ProcessPool implementation
+            num_jobs = len(convert_video_params)
+            with ProcessPoolExecutor(max_workers=num_jobs) as executor:
+                jobs = [
+                    executor.submit(convert_video, *params)
+                    for params in convert_video_params
+                ]
+                for job in as_completed(jobs):
+                    try:
+                        result = job.result()
+                        logging.info("FFmpeg job completed:", result)
+                    except Exception as e:
+                        logging.info("Error:", e)
```

Comment:
This is swallowing errors. I strongly think any error should cause the entire job to error.